qiskit/primitives/__init__.py

```diff
@@ -46,11 +46,13 @@
 .. autosummary::
    :toctree: ../stubs/
 
+   BaseResult
    EstimatorResult
    SamplerResult
 """
 
 from .base_estimator import BaseEstimator
+from .base_result import BaseResult
```

Comment:
I am neutral about putting this class in `__init__.py`. I don't think it is necessary if there is no use case.
Comments are welcome.